src/System.Private.Xml/src/System/Xml/Serialization/ReflectionXmlSerializationReader.cs

```diff
@@ -59,155 +62,511 @@ protected override void InitIDs()
-                    if (pi != null && typeof(IList).IsAssignableFrom(pi.PropertyType)
-                        && (pi.SetMethod == null || !pi.SetMethod.IsPublic))
+                    string name = element.Name;
+                    string ns = element.Form == XmlSchemaForm.Qualified ? element.Namespace : string.Empty;
+                    if (!XmlNodeEqual(Reader, name, ns))
                     {
-                        var getOnlyList = (IList)pi.GetValue(o);
+                        throw CreateUnknownNodeException();
+                    }
+                }
 
-                        var valueList = value as IList;
-                        if (valueList != null)
-                        {
-                            foreach (var i in valueList)
-                            {
-                                getOnlyList.Add(i);
-                            }
-                        }
-                        else
+                isEmptyWrapper = Reader.IsEmptyElement;
+                Reader.ReadStartElement();
+            }
+
+            Member[] members = new Member[mappings.Length];
+            for (int i = 0; i < mappings.Length; i++)
+            {
+                int index = i;
+                MemberMapping mapping = mappings[index];
+                var member = new Member(mapping);
+                member.Source = (value) => p[index] = value;
+                members[index] = member;
+                if (mapping.CheckSpecified == SpecifiedAccessor.ReadWrite)
+                {
+                    string nameSpecified = mapping.Name + "Specified";
+                    for (int j = 0; j < mappings.Length; j++)
+                    {
+                        if (mappings[j].Name == nameSpecified)
                         {
-                            getOnlyList.Add(value);
+                            int indexOfSpecifiedMember = j;
+                            member.CheckSpecifiedSource = (value) => p[indexOfSpecifiedMember] = value;
+                            break;
                         }
                     }
-                    else if (member.Collection != null)
+                }
+
+            }
+
+            Fixup fixup = WriteMemberFixupBegin(members, p);
+            if (members.Length > 0 && members[0].Mapping.IsReturnValue)
+            {
+                IsReturnValue = true;
+            }
+
+            List<CheckTypeSource> checkTypeHrefSource = null;
+            if (!hasWrapperElement && !writeAccessors)
+            {
+                checkTypeHrefSource = new List<CheckTypeSource>();
+            }
+
+            Reader.MoveToContent();
+            while (Reader.NodeType != XmlNodeType.EndElement && Reader.NodeType != XmlNodeType.None)
+            {
+                UnknownNodeAction unrecognizedElementSource;
+                if (checkTypeHrefSource == null)
+                {
+                    unrecognizedElementSource = (_) => UnknownNode(p);
```

Comment:
>UnknownNode(p); [](start = 55, length = 15)

Is this initialization code which runs once per serializer instantiation or is it run on every serialization? If the latter, then this is allocating a closure each time.